scripts/test.js

```diff
@@ -69,5 +70,21 @@ if (
   argv.push(hasSourceControl ? '--watch' : '--watchAll');
 }
 
-
-jest.run(argv);
+const scopeKey = 'scope';
+const scopePropIndex = argv.indexOf(`--${scopeKey}`);
+let args = argv;
+const defaultScopeFile = 'vestri.json';
+const parsedProps = yargs.parse(process.argv);
+const testDirPath = `${process.cwd()}/src/tests/e2e/scopes`;
+if (scopePropIndex !== -1) {
+  args = argv.filter(arg => !(arg === `--${scopeKey}` || arg === parsedProps[scopeKey]));
+}
+fs.readdir(testDirPath, (err, files) => {
+  if (err) {
+    throw Error('Scopes directory not found');
+  }
+  const scopePropFile = files.find(file => file === `${parsedProps[scopeKey]}.json`);
+  const scopeFile = scopePropFile || defaultScopeFile;
+  process.env.SCOPE_DIR = `./scopes/${scopeFile}`;
+  jest.run(args);
```

Comment:
Excellent flow! I'll keep track to write some documentation for supporting this 👍